utils/emmylua/wesnoth.lua

```diff
@@ -0,0 +1,272 @@
+---@meta
+
+wesnoth = {}
+
+---@return "'Game Lua Kernel'" | "'Mapgen Lua Kernel'" | "'Application Lua Kernel'"
+function wesnoth.kernel_type() end
+
+---@param path string
+---@return ...
+function wesnoth.dofile(path, ...) end
+
+---@class tstring : string
+
+---Constructs a textdomain, which can be called to create translatable strings.
+---@param domain string The textdomain name
+---@return fun(str:string, str_plural?:string, n?:number):tstring
+function wesnoth.textdomain(domain) end
+
+---Logs a message to the console
+---@param logger "'info'"|"'debug'"|"'warning'"|"'error'"|"'wml'"
+---@param message string
+---@param in_chat boolean
+---@overload fun(message:string, in_chat:boolean)
+function wesnoth.log(logger, message, in_chat) end
+
+---Simulate combat between two units_MP
```

Comment:
The `units_MP` looks like a cut-and-paste typo - there's a local variable of that name somewhere, used for "units with movement points".

Reply:
I think it's probably an auto-complete issue. Sometimes the auto-complete just accepts the first suggestion without me noticing.